.github/workflows/reusable-prober.yml

```diff
@@ -113,76 +110,40 @@ jobs:
 
-      - name: Check expiration
-        id: check
-        run: |
-          if [ "${{ inputs.tuf_root_url }}" != "" ]; then
-            echo "::warning::fetching root.json from ${{ inputs.tuf_root_url }}"
-            curl -Lo root.json --user-agent "sigstore-prober" ${{ inputs.tuf_root_url }}
-            export ROOT_PATH=root.json
-          else
-            export ROOT_PATH=$GITHUB_WORKSPACE/root-signing/${{ inputs.tuf_root_path }}
-          fi
-          export PD_DEDUP_KEY_FILE=$(mktemp)
-
-          echo "::group::tuf-repo-cdn 2 day check"
-          export EXPIRY=$(date -d '+2 days' '+%s')
-          verify repository --repository ${{ inputs.tuf_repo }} --root ${ROOT_PATH} --valid-until ${EXPIRY} --dedup-key-file ${PD_DEDUP_KEY_FILE} || (echo "dedup_key=$(cat ${PD_DEDUP_KEY_FILE})" >> $GITHUB_OUTPUT ; exit 1)
-          echo "::endgroup::"
-
-          echo "::group::tuf-repo-cdn 15 day check"
-          export EXPIRY=$(date -d '+15 days' '+%s')
-          echo "Verifying root valid within 15 days..."
-          verify repository --repository ${{ inputs.tuf_repo }} --root ${ROOT_PATH} --valid-until ${EXPIRY} --role root.json --role targets.json --dedup-key-file ${PD_DEDUP_KEY_FILE} || (echo "dedup_key=$(cat ${PD_DEDUP_KEY_FILE})" >> $GITHUB_OUTPUT ; exit 1)
-          echo "::endgroup::"
-
-          # For preprod/staging TUF bucket
-          echo "::group::tuf-preprod-repo-cdn 2 day check"
-          export EXPIRY=$(date -d '+2 days' '+%s')
-          verify repository --repository ${{ inputs.tuf_preprod_repo }} --root ${ROOT_PATH} --valid-until ${EXPIRY} --dedup-key-file ${PD_DEDUP_KEY_FILE} || (echo "dedup_key=$(cat ${PD_DEDUP_KEY_FILE})" >> $GITHUB_OUTPUT ; exit 1)
-          echo "::endgroup::"
-
-          echo "::group::tuf-preprod-repo-cdn 15 day check"
-          export EXPIRY=$(date -d '+15 days' '+%s')
-          echo "Verifying root valid within 15 days..."
-          verify repository --repository ${{ inputs.tuf_preprod_repo }} --root ${ROOT_PATH} --valid-until ${EXPIRY} --role root.json --role targets.json --dedup-key-file ${PD_DEDUP_KEY_FILE} || (echo "dedup_key=$(cat ${PD_DEDUP_KEY_FILE})" >> $GITHUB_OUTPUT ; exit 1)
-          echo "::endgroup::"
+      - name: Verify preprod TUF repository state (and the upgrade from prod)
+        if: ${{ inputs.enable_staging == false }}
+        uses: theupdateframework/tuf-on-ci/actions/test-repository@ebf63d46b20e8c019e7752f8fb57c28931f96cfc # v0.9.0
+        with:
+          metadata_url: ${{ inputs.tuf_preprod_repo }}
+          update_base_url:  ${{ inputs.tuf_repo }}
+          valid_days: 2
+          offline_valid_days: 15
+          metadata_dir: tuf/preprod/
+          compare_source: false
 
       - name: Set messages
         id: msg
         if: success() || failure()
         run: |
+          # calculate a dedup key based on all TUF metadata the test client saw
+          echo "dedup_key=$(sha256sum tuf/*/* | sha256sum | cut -d " " -f 1)" >> $GITHUB_OUTPUT
```

Comment:
This does not print out the dedup key or list of files... those could be useful for debugging if that's needed